x11rb-protocol/src/parse_display/mod.rs

```diff
@@ -59,15 +60,25 @@ pub fn parse_display(dpy_name: Option<&str>) -> Option<ParsedDisplay> {
 pub fn parse_display_with_file_exists_callback(
     dpy_name: Option<&str>,
     file_exists: impl Fn(&str) -> bool,
-) -> Option<ParsedDisplay> {
+) -> Result<ParsedDisplay, DisplayParsingError> {
     // If no dpy name was provided, use the env var. If no env var exists, return None.
     match dpy_name {
         Some(dpy_name) => parse_display_impl(dpy_name, file_exists),
-        None => parse_display_impl(&std::env::var("DISPLAY").ok()?, file_exists),
+        None => match std::env::var("DISPLAY") {
```

Comment:
I just noticed this. We cannot use `std::env::var` without `std`.

Reply:
I noticed that too while writing this change and chose to ignore it for now. Since CI passes despite this clearly accessing `std`, to me that means we need better CI. Random google result is https://blog.dbrgn.ch/2019/12/24/testing-for-no-std-compatibility/ but for today I don't have more time to work on this, so perhaps next weekend.

Reply:
Ahhh! This file has `#![cfg(feature = "std")]` in line 3. So... apparently I have been lots of complicated things for nothing.

Reply:
Whoopsie daisy 🫠 I should've spotted that, I think I'm the one that wrote it!